src/main/java/hudson/remoting/Response.java

```diff
@@ -51,14 +54,19 @@
     final RSP returnValue;
     final EXC exception;
 
-    Response(int id, int lastIoId, RSP returnValue) {
+    private transient long totalTime;
+    private transient Request<?, ?> request;
```

Comment:
Maybe it should a WeakReference here, needs discussion. Otherwise the GC won't be able to collect Requests until the response it fully processed.  Fine for diagnostics, but may impact the overall performance on heavily loaded Jenkins masters.

Reply:
I would also suggest caching the name of the request. So the implementation could be similar to ChannelStateException (see https://github.com/jenkinsci/remoting/pull/235/files)

Reply:
Adding annotation to the field would not hurt also